graph/encoding/dot/encode.go

```diff
@@ -265,11 +282,44 @@ func (p *printer) print(g graph.Graph, name string, needsIndent, isSubgraph bool
 			p.buf.WriteByte(';')
 		}
 	}
-	p.closeBlock("}")
 
+	p.printBackMatter()
+
+	return nil
+}
+
+func (p *printer) printFrontMatter(name string, needsIndent, isSubgraph, isDirected, isStrict bool) error {
+	p.buf.WriteString(p.prefix)
+	if needsIndent {
+		for i := 0; i < p.depth; i++ {
+			p.buf.WriteString(p.indent)
+		}
+	}
+
+	if !isSubgraph && isStrict {
+		p.buf.WriteString("strict ")
+	}
+
+	if isSubgraph {
+		p.buf.WriteString("sub")
+	} else if isDirected {
+		p.buf.WriteString("di")
+	}
+	p.buf.WriteString("graph")
+
+	if name != "" {
+		p.buf.WriteByte(' ')
+		p.buf.WriteString(name)
+	}
+
+	p.openBlock(" {")
 	return nil
 }
 
+func (p *printer) printBackMatter() {
```

Comment:
Do we need this?

Reply:
I went back and forth on this; and I'm not sure... It felt weird to _not_ have it given that it starts with a common printFrontMatter(), it felt unsymmetrical to unroll it and print a seemingly unmatched `}` here. But, as I think you are implying it feels superfluous to print a single `}`. So, I'm not sure and I'll defer to whatever you think is best.

Reply:
Can you remove it.